### PR TITLE
Add infra SNS topics

### PR DESCRIPTION
**Pull request by cat5inthecradle**
In an effort to make alerting more useful, we're looking to move some alerts from `CDO-<urgency>` topics to `Infra-<urgency>` topics. User-impacting issues would still notify the DOTD, while lower-level issues or symptoms will directly notify Infra. This PR creates the SNS topics to enable this work. These can be integrated with PagerDuty to page the Infra on-call engineer.

Other options were considered, but the final version of this PR implements these three topics in each deployed environment. They can be subscribed to by external services.

## Deployment strategy

Squash and merge. This will deploy automatically once merged.

## Follow-up work

- Integrate these topics with PagerDuty and test the integration
- Reconfigure the Firehose Put Requests alarm to notify Infra-WorkdayUrgent instead of CDO-WorkdayUrgent

## Privacy

n/a

## Security

SNS topics may contain sensitive information, like stack traces. They also may reveal information about our infrastructure. Permissions to read SNS topics are granted via AWS IAM.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by sureshc** [comment]
I think the Alerting Stack was an experiment that we never really embraced. It might have been created when we were starting to use scheduled Lambdas for some infrastructure processes (such as backup of database to offsite) and we wanted the same type of HoneyBadger alerting we get from a cron job that fails on `production-daemon` with a scheduled Lambda.

The goal here, I think, is to add monitoring of internal infrastructure components our production Stack to alert Infrastructure engineers. Some alerts will be urgent, and some only workday urgent, or low priority.  I lean towards putting this logic in our existing main stack's [Alarms component](https://github.com/code-dot-org/code-dot-org/blob/d692f38fc121a0e864948d127868652c62c9dd6f/aws/cloudformation/components/alarms.yml.erb) wrapped inside `if production` statements.

**Comment by cat5inthecradle** [comment]
> I lean towards putting this logic in our existing main stack's [Alarms component](https://github.com/code-dot-org/code-dot-org/blob/d692f38fc121a0e864948d127868652c62c9dd6f/aws/cloudformation/components/alarms.yml.erb) wrapped inside `if production` statements.

My only hesitation with that is the potentially very unlikely scenario where we're provisioning new environments, and we have referenced these production-only resources in a lower environment before they actually exist in the higher environment. We definitely have things in lower environments that warrant alerting.

Another option... we do what you suggest, but don't wrap it in `if production`. We create topics in every environment. They can funnel together into one stream at the PagerDuty level. If I drop the "TopicName" property, these would get named something like `autoscale-prod-Infra-WorkdayUrgent-er7y5t4ihuer98y4t` and `test-Infra-WorkdayUrgent-u8r3u982oi4r`. (we might be able to use the DisplayName property to tidy things up a bit). This would avoid any potential chicken/egg situations, would make the topics easily referable within the templates, and would guarantee environment distinction even if the message contents didn't properly distinguish the environment. We could also tune alerting per-environment more easily, like upgrading production workday urgent messages to page 24 hours during critical time periods without also escalating other environments (via PagerDuty).

**Comment by cat5inthecradle** [comment]
We have an example of per-env topics already, at the top of Alarms.yml. These two topics only go to slack, via our AWS Chatbot configuration: https://us-east-2.console.aws.amazon.com/chatbot/home?region=us-east-1#/chat-clients/slack/workspaces/T039SAH7W/configurations/slack-infra

**Comment by sureshc** [comment]
Stack-specific SNS Topics makes sense to me with conditional logic to determine whether they route events to PagerDuty or to Slack.

File: aws/cloudformation/alerting.yml.erb [hunk 1]
```diff
@@ -1,13 +1,12 @@
-# Creates a SNS topic and Lambda which publish events to Honeybadger. This allows for any Cloudwatch Alarm
```
Comment (cat5inthecradle) [inline aws/cloudformation/alerting.yml.erb:1]: I originally considered placing the resources in this template. These edits simply rearrange comments. This stack is updated manually, and will not be updated automatically upon deployment.

**Review by Hamms — COMMENTED** [review]
This change makes sense to me as a first step, but I am not intimately familiar with the details of this space and so can't comment on whether or not these being stack-specific would be better

**Review by Hamms — APPROVED** [review]

**Comment by cat5inthecradle** [comment]
This will be stack specific, it's perhaps overkill, but I'll find out pretty quick whether it's too much of a hassle.

**Pull request merged**